app/Http/Transformers/SignupTransformer.php

```diff
@@ -48,12 +56,31 @@ public function transform(Signup $signup)
      * Include posts
      *
      * @param \Rogue\Models\Signup $signup
+     * @param \League\Fractal\ParamBag|null
      * @return \League\Fractal\Resource\Collection
      */
-    public function includePosts(Signup $signup)
+    public function includePosts(Signup $signup, ParamBag $params = null)
     {
-        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
-        $post = $signup->visiblePosts;
+        if ($params === null) {
+            // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
+            $post = $signup->visiblePosts;
+
+            return $this->collection($post, new PostTransformer);
+        }
+
+        // Optional params validation
+        $usedParams = array_keys(iterator_to_array($params));
+
+        if ($invalidParams = array_diff($usedParams, $this->validParams)) {
+            throw new \Exception(sprintf(
+                'Invalid param(s): "%s". Valid param(s): "%s"',
+                implode(',', $usedParams),
+                implode(',', $this->validParams)
+            ));
+        }
+
+        // Only allow an admin or the user who owns the signup to see the signup's unapproved posts by type.
+        $post = $signup->visiblePosts($params->get('type'));
```

Comment:
I'd also like to double-check that we're okay calling `visiblePosts` as a method here, since usually relationships are accessed as a property. Let's make sure we're not accidentally adding additional SQL queries!

Reply:
ah yes so much cleaner!